src/frontend-samples/swiping-viewport/Divider.tsx

```diff
@@ -0,0 +1,158 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import * as React from "react";
+import "./Divider.scss";
+
+export interface DividerComponentState {
+  left: number;
+  bounds: ClientRect;
+  // children updates
+}
+
+export interface DividerComponentProps {
+  bounds: ClientRect;
+  menuChildren?: React.ReactNode;
+  leftChildren?: React.ReactNode;
+  rightChildren?: React.ReactNode;
+  onDragged?: (leftPanelWidth: number, rightPanelWidth: number) => void;
+  buffer?: number;
+  sideL?: number;
+  sideR?: number;
+}
+
+export class DividerHandleComponent extends React.Component<{}, {}> {
+  public render() {
+    return <div className={"grab-handle"} id={"grabber-div"}></div>;
+  }
+}
+
+export class DividerComponent extends React.Component<DividerComponentProps, {}> {
+  public state: DividerComponentState;
+
+  private _oldPosition: number = 0;
+  private _elem?: HTMLElement;
+  private _container: HTMLElement | null = null;
+
+  private get _buffer(): number {
+    return undefined === this.props.buffer ? 0 : this.props.buffer;
+  }
+  private get _width(): number {
+    let width: number = 0;
+    if (null === this._container)
+      return width;
+    const widthStr = window.getComputedStyle(this._container)?.getPropertyValue("width");
+    if (undefined !== widthStr && widthStr)
+      width = parseInt(widthStr, 10);
+    return width;
+  }
+  private limitToBounds(n: number): number {
+    n = Math.min(n, this.props.bounds.right - (this._elem!.clientWidth + this._buffer));
+    n = Math.max(n, this.props.bounds.left + this._buffer);
+    return n;
+  }
+
+  constructor(props: DividerComponentProps, context: any) {
```

Comment:
Remove the "context" argument.  The code editor in the showcase reports this as a "problem" at runtime.